src/proteus/dswx_hls.py

```diff
@@ -1772,14 +1810,46 @@ def _load_hls_from_file(filename, image_dict, offset_dict, scale_dict,
 
     metadata = layer_gdal_dataset.GetMetadata()
 
+    # read image
+    if flag_debug:
+        logger.info('reading in debug mode')
+        image = layer_gdal_dataset.ReadAsArray(
+            xoff=0, yoff=0, xsize=1000, ysize=1000)
+    else:
+        image = layer_gdal_dataset.ReadAsArray()
+
+
+    # read `fill_value`
+    if fill_value is None and '_FillValue' in metadata.keys():
+        fill_value = float(metadata['_FillValue'])
+    elif fill_value is None:
+        fill_value = -9999
+
+    # create/update `invalid_ind_array`
+    # (invalid pixels are cumulative)
```

Comment:
Coming in with fresh eyes, it was confusing to me that this function's primary purpose seems to be to read in a single band and update the `image_dict` with the information for just that one HLS band. (tbh, I thought this function's purpose was to read in all bands from the product!)

If you have a chance, it could be helpful to make these updates:

1) update the name of the function to something like, `_load_hls_band_from_file(..)`

2) update the input paramter `key` to being something like `band` (this is what the docstring says)

3) update the docstring's description for `image_dict` to be more descriptive, something like:
```
`Image dictionary that will store HLS band array and information about the bands.
The following keys in `image_dict` will be created/updated based on the
information found in the HLS file for the given band: `invalid_ind_array`, 
`geotransform`, `projection`, `length`, `width`.
```
(I might be missing some keys)

4) update this comment to be something like:

```suggestion
    # create/update `invalid_ind_array` with invalid pixels in this band.
    # (invalid pixels are cumulative for all bands)
```